### PR TITLE
fix: use arguments to `\par` if available

### PR DESCRIPTION
**Pull request by Nerixyz**
`\par` accepts the heading as the first argument.

Fixes https://github.com/cppalliance/mrdocs/issues/765.

Aside: I noticed that the tests for the adoc and HTML generators are taking much longer than the XML tests (also visible in CI).

### CODE REVIEW

File: test-files/golden-tests/para-3.adoc [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+[source,cpp,subs="verbatim,replacements,macros,-callouts"]
+----
+void
+my&lowbar;function();
+----
+
+=== Description
+
+
+=== A heading&period;
```
Comment (Nerixyz) [inline test-files/golden-tests/para-3.adoc:35]: This seems wrong - the "A heading." should probably be one level "lower" than "Description" (but that's not really part of this PR, I suppose).

File: test-files/golden-tests/para-3.html [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+<pre>
+<code class="source-code cpp">
+void
+my_function();
+</code>
+</pre>
+</div>
+<div>
+<h3>Description</h3>
+<h3>A heading.</h3>
```
Comment (Nerixyz) [inline test-files/golden-tests/para-3.html:47]: Same here.

File: test-files/golden-tests/para-3.adoc (hunk 1 above)
Comment (alandefreitas) [reply test-files/golden-tests/para-3.adoc:35]: Yes. This seems reasonable enough. There's a lot to think about regarding how this overview is structured.
